src/main/java/com/google/collegeplanner/servlets/PlannerServlet.java

```diff
@@ -29,35 +36,226 @@
 /** Servlet that returns the mutli-semester plan for the given courses.*/
 @WebServlet("/api/planner")
 public class PlannerServlet extends HttpServlet {
+  private HashMap<String, Integer> indegree; // Number of prerequisites for a course
+  private HashMap<String, HashSet<String>>
+      nextCourses; // Courses that come after the given course in the graph
+  private HashMap<String, HashSet<String>> corequisites;
+  private HashMap<String, Integer> heights; // height of a course in the graph
+  private HashMap<String, Integer> credits; // Number of credits for each course
+  private LinkedList<String> courseList;
+  private int totalCredits;
+
   /**
-   * Organizes courses into the given number of semesters
+   * Organizes courses from POST request into a given number of semesters
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO(ramyabuva): Implement algorithm to split into the number of semesters
     JSONObject body;
+    int numSemesters;
+    JSONArray selectedClasses;
     try {
       body = getBody(request);
-    } catch (ParseException | NullPointerException e) {
+      numSemesters = Integer.parseInt((String) body.get("semesters"));
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (NumberFormatException | ClassCastException | ParseException | NullPointerException e) {
       respondWithError(
           "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
       return;
     }
-    String strClasses = (String) body.get("selectedClasses");
-    // Returns the courses in a single semester
-    String[][] courses = {strClasses.split(",")};
+
+    constructGraphs(selectedClasses);
+
+    ArrayList<ArrayList<String>> semesters = getPlan(numSemesters);
+    JSONObject coursePlan = new JSONObject();
+    coursePlan.put("semester_plan", semesters);
     response.setContentType("applications/json;");
-    response.getWriter().println(new Gson().toJson(courses));
+    response.getWriter().println(new Gson().toJson(coursePlan));
+  }
+
+  /**
+   * Creates and returns a valid plan for the given number of semesters
+   * @param numSemesters The number of semesters remaining
+   */
+  private ArrayList<ArrayList<String>> getPlan(int numSemesters) {
```

Comment:
i think you have a bad commit in your history that is adding this code. this shouldn't be showing up in your diff.